plenum/server/replica.py

```diff
@@ -198,31 +198,34 @@ def preprepare_batch(self, pp: PrePrepare):
         """
         After pp had validated, it placed into _preprepared list
         """
-        if pp in self.consensus_data.preprepared:
+        pp_batch_id = batch_id(pp)
+        if pp_batch_id in self.consensus_data.preprepared:
             raise LogicError('New pp cannot be stored in preprepared')
-        if self.consensus_data.checkpoints and pp.ppSeqNo < self.consensus_data.last_checkpoint.seqNoEnd:
+        if self.consensus_data.checkpoints and pp.pp_seq_no < self.consensus_data.last_checkpoint.seqNoEnd:
```

Comment:
I think it should either stay as `pp.ppSeqNo` or be renamed to `pp_batch_id.pp_seq_no`